src/tests/tribol_enzyme_merged_jacobian.cpp

```diff
@@ -0,0 +1,656 @@
+        if ( diff > delta ) {
+          std::cout << "  (" << i << ", " << j << ") : Diff: " << diff << "   Enzyme: " << dfdp_enzyme( i, j )
+                    << "   FD: " << dfdp_fd( i, j ) << std::endl;
+        }
+        EXPECT_NEAR( dfdp_enzyme( i, j ), dfdp_fd( i, j ), delta );
+      }
+    }
+  }
+};
+
+TEST_F( EnzymeMergedJacobianTest, FiniteDiffCheckShifted2x2Meshes )
+{
+  constexpr auto num_xel_mesh0 = 2;
+  constexpr auto num_yel_mesh0 = 2;
+  constexpr auto num_xel_mesh1 = 2;
+  constexpr auto num_yel_mesh1 = 2;
+  constexpr auto el_width = 1.0;
+  constexpr auto el_height = 1.0;
+  constexpr auto xy_shift = 0.1;
+
+  constexpr auto mesh0_bdry_attrib = 7;
+  constexpr auto mesh1_bdry_attrib = 8;
+
+  // clang-format off
+  auto mesh = shared::MeshBuilder::Unify({
+    shared::MeshBuilder::CubeMesh(num_xel_mesh0, num_yel_mesh0, 1)
+      .scale({el_width, el_width, el_height})
+      // shift down 0.5% height of element (1% elem thickness interpenetration)
+      .translate({0.0, 0.0, -0.005 * el_height})
+      // change the mesh0 boundary attribute from 1 to 7
+      .updateBdrAttrib(1, mesh0_bdry_attrib),
+    shared::MeshBuilder::CubeMesh(num_xel_mesh1, num_yel_mesh1, 1)
+      .scale({el_width, el_width, el_height})
+      // shift down 99.5% height of element (1% elem thickness interpenetration)
+      .translate({0.0, 0.0, -0.995 * el_height})
+      // shift x and y so the element edges are not overlapping
+      .translate({xy_shift, xy_shift, 0.0})
+      // change the mesh1 boundary attribute from 6 to 8
+      .updateBdrAttrib(6, mesh1_bdry_attrib)
+  });
+  // clang-format on
+
+  RunJacobianTest( mesh, nullptr, nullptr, mesh0_bdry_attrib, num_xel_mesh0, num_yel_mesh0, mesh1_bdry_attrib,
+                   num_xel_mesh1, num_yel_mesh1 );
+}
+
+TEST_F( EnzymeMergedJacobianTest, FiniteDiffCheckShifted1x1Meshes )
+{
+  constexpr auto num_xel_mesh0 = 1;
+  constexpr auto num_yel_mesh0 = 1;
+  constexpr auto num_xel_mesh1 = 1;
+  constexpr auto num_yel_mesh1 = 1;
+
+  constexpr auto mesh0_bdry_attrib = 6;
+  constexpr auto mesh1_bdry_attrib = 7;
+
+  constexpr double eps = 1.0e-7;
+  constexpr double xy_shift = eps * 10.0;
+  // clang-format off
+  auto mesh = shared::MeshBuilder::Unify({
+    shared::MeshBuilder::CubeMesh(1, 1, 1),
+    shared::MeshBuilder::CubeMesh(1, 1, 1)
+      // shift up 99.9% height of element
+      .translate({0.0, 0.0, 0.999})
+      // shift x and y so the element edges are not overlapping
+      .translate({xy_shift, xy_shift, 0.0})
+      // change the mesh1 boundary attribute from 1 to 7
+      .updateBdrAttrib(1, 7)
+      // change the mesh1 boundary attribute from 6 to 8
+      .updateBdrAttrib(6, 8)
+  });
+  // clang-format on
+
+  RunJacobianTest( mesh, nullptr, nullptr, mesh0_bdry_attrib, num_xel_mesh0, num_yel_mesh0, mesh1_bdry_attrib,
+                   num_xel_mesh1, num_yel_mesh1 );
+}
+
+TEST_F( EnzymeMergedJacobianTest, FiniteDiffCheckShifted1x1MeshesV2 )
```

Comment:
Maybe I missed it, but do you want to check a warped mesh configuration like you do in tribol_enzyme_jacobian? Seems like adding the normal contribution would matter for warped elements?

Reply:
Good call, test has been added.